src/bin/api.rs

```diff
@@ -33,6 +33,12 @@ async fn main() -> std::io::Result<()> {
         println!("Babamul API endpoints are DISABLED");
     }
 
+    // read the port from env var BOOM_API__PORT, default to 4000
+    let port: u16 = std::env::var("BOOM_API__PORT")
+        .unwrap_or_else(|_| "4000".to_string())
+        .parse()
+        .expect("Invalid port number");
+
```

Comment:
Instead of reading it here, maybe it makes sense to put it into the `ApiConfig` struct so it's read from env vars automatically.